libglusterfs/src/unittest/mem_pool_unittest.c

```diff
@@ -34,8 +35,8 @@ typedef struct __attribute__((packed)) {
     uint32_t type;
     size_t size;
     xlator_t *xl;
-    uint32_t header_magic;
-    uint8_t pad[8];
+    gf_mem_magic_t header_magic;
+    void *data[];
 } mem_header_t;
```

Comment:
I think this structure is not correct for systems that don't allow unaligned accesses. Being it `packed` means that `size` may be unaligned on architectures where `size_t` is 8 bytes.

I think we should remove the `packed` attribute and adjust the fields to avoid internal padding.

Reply:
It should not be packed, no. There's no packed attribute on this, so the compiler will pad it so it's aligned for the architecture.

Reply:
Bear in mind, it's the architectures that don't allow unaligned access that this patch is fixing. Gluster at present does not run on Linux/SPARC64, it crashes with SIGBUS. With this patchset it does not.

Reply:
The structure is actually packed (GitHub doesn't allow me to select lines outside of the modified ones). The Actual definition is:

```c
typedef struct __attribute__((packed)) {
    uint32_t type;
    size_t size;
    xlator_t *xl;
    gf_mem_magic_t header_magic;
    void *data[];
} mem_header_t;
```

So it's packed and the first field is a 4 bytes value, and the second one could be an 8 bytes value in some architectures.

Reply:
Are you sure the `packed` attribute doesn't interfere with SPARC's memory alignment requirements ?

Reply:
How did I miss the packed attribute? Sorry, there must be something wrong with my eyes. I'll remove that.

Reply:
What's the reason to to have a slightly different header definition in the unit test btw? Why not just include and use the original?